### PR TITLE
Android build and export for the mono module

### LINKED ISSUE #20267: Mono/C# support on Android
We lacked an issue to keep track of Android export support for Mono/C# projects, so here's one.

@neikeq What are the needed steps to get this to work?

### PR DESCRIPTION
**Pull request by neikeq**
With this PR we can build Godot with Mono for Android and export games (both with pre-built templates and the new custom build that was added recently).

I will be updating the [Compiling with Mono](https://docs.godotengine.org/en/latest/development/compiling/compiling_with_mono.html) page in the following days.

*Bugsquad edit:* Fixes #20267.

### CODE REVIEW

File: modules/mono/build_scripts/build.mono.android.sh [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ANDROID_TOOLCHAIN_CACHE_DIR=${ANDROID_TOOLCHAIN_CACHE_DIR:-${HOME}/android-archives}
+ANDROID_TOOLCHAIN_PREFIX=${ANDROID_TOOLCHAIN_PREFIX:-${ANDROID_TOOLCHAIN_DIR}/toolchains}
+
+export ANDROID_TOOLCHAIN_DIR ANDROID_TOOLCHAIN_CACHE_DIR ANDROID_TOOLCHAIN_PREFIX
+
+make -C sdks/builds provision-android && make -C sdks/android accept-android-license
+make -C sdks/builds provision-mxe
+make -C sdks/builds archive-android NINJA= IGNORE_PROVISION_ANDROID=1 IGNORE_PROVISION_MXE=1
+
+function strip_libs {
```
Comment (aaronfranke) [inline modules/mono/build_scripts/build.mono.android.sh]: Ditto

File: modules/mono/build_scripts/build.godot.android.sh [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+SCONS_JOBS=${SCONS_JOBS:-2}
+SCONS_MONO_STATIC=${SCONS_MONO_STATIC:-yes}
+SCONS_VERBOSE=${SCONS_VERBOSE:-yes}
+
+: ${MONO_SOURCE_ROOT:?Variable MONO_SOURCE_ROOT not set or empty}
+
+# The SCons script expects an absolute path
+MONO_SOURCE_ROOT="$( cd "${MONO_SOURCE_ROOT}" >/dev/null 2>&1 && pwd )"
+
+function scons_build {
```
Comment (aaronfranke) [inline modules/mono/build_scripts/build.godot.android.sh]: To comply with POSIX standards, shell functions should be declared without the word "function" and with parenthesis:

    scons_build() {

File: modules/mono/build_scripts/build.mono.android.sh (hunk 1 above)
Comment (neikeq) [reply modules/mono/build_scripts/build.mono.android.sh]: Thanks!

**Comment by akien-mga** [comment]
Needs a rebase, otherwise I think we can merge it as is and see what needs to be improved regarding build scripts/integration with the usual SCons workflow later on.

From reading the docs at godotengine/godot-docs#2493, it seems like it's well integrated already.

**Comment by akien-mga** [comment]
@hpvb Relevant change for the buildsystem, apart from the need to build the Mono SDK beforehand: the mono prefix should now be passed via the `mono_prefix` SCons option instead of env variables.

**Comment by neikeq** [comment]
Got rid of the bash scripts. The one for Mono is on the docs page now. If anyone is interested in the other one: https://gist.github.com/neikeq/b8b62075f0032efd67355ce7e7246266

**Comment by akien-mga** [comment]
Thanks a lot!

**Comment by nightblade9** [comment]
This is awesome! Fantastic work!

How big is a "hello world" Android C# build of Godot?

**Comment by damccull** [comment]
I'm very excited about this! I can't wait to see it in a release.

**Pull request merged**